File: problems/2025/09-15/string-stack/solution.js
function stringStack(pat, tar) {
  let j = 0;
  for (let i = 0; i < pat.length; i++) {
    if (j < tar.length && pat[i] === tar[j]) {
      j++;
    }
  }
  return j === tar.length;
}

console.log(stringStack("geuaek", "geek"));    // true
console.log(stringStack("agiffghd", "gfg"));   // true
console.log(stringStack("ufahs", "aus"));      // false

File: problems/2025/09-15/string-stack/solution.py
def string_stack(pat: str, tar: str) -> bool:
    j = 0
    for ch in pat:
        if j < len(tar) and ch == tar[j]:
            j += 1
    return j == len(tar)

if __name__ == "__main__":
    print(string_stack("geuaek", "geek"))      # True
    print(string_stack("agiffghd", "gfg"))     # True
    print(string_stack("ufahs", "aus"))        # False
